### ISSUE TITLE
Do not override the SSL settings in a MB_DB_CONNECTION_URI (or any JDBC URI) for PostgreSQL

### ISSUE THREAD

**Issue by robdaemon** [metabase/metabase#13796]
**Describe the bug**
You cannot connect to a PostgreSQL instance that requires client certificate authentication.

**Logs**

```
2020-11-13 17:07:24,711 ERROR metabase.core :: Metabase Initialization FAILED                                                                                                                                                                               java.lang.IllegalArgumentException: No matching clause: FATAL: connection requires a valid client certificate                                                                                                                                                 Location: File: auth.c, Routine: ClientAuthentication, Line: 397                                                                                                                                                                                            Server SQLState: 28000                                                                                                                                                                                                                                            at metabase.driver.postgres$eval64542$fn__64543.invoke(postgres.clj:50) ~[?:?]                                                                                                                                                                              at clojure.lang.MultiFn.invoke(MultiFn.java:234) ~[clojure-1.10.1.jar:?]                                                                                                                                                                                    at metabase.driver.util$can_connect_with_details_QMARK_.invokeStatic(util.clj:40) ~[?:?]                                                                                                                                                                    at metabase.driver.util$can_connect_with_details_QMARK_.doInvoke(util.clj:23) ~[?:?]                                                                                                                                                                        at clojure.lang.RestFn.invoke(RestFn.java:442) ~[clojure-1.10.1.jar:?]                                                                                                                                                                                      at clojure.lang.Var.invoke(Var.java:393) ~[clojure-1.10.1.jar:?]                                                                                                                                                                                            at metabase.db$eval24852$verify_db_connection__24861$fn__24864$fn__24865.invoke(db.clj:316) ~[?:?]                                                                                                                                                          at metabase.db$eval24852$verify_db_connection__24861$fn__24864.invoke(db.clj:315) ~[?:?]                                                                                                                                                                    at metabase.db$eval24852$verify_db_connection__24861.invoke(db.clj:307) ~[?:?]                                                                                                                                                                              at metabase.db$eval24852$verify_db_connection__24861$fn__24862.invoke(db.clj:310) ~[?:?]                                                                                                                                                                    at metabase.db$eval24852$verify_db_connection__24861.invoke(db.clj:307) ~[?:?]                                                                                                                                                                              at metabase.db$setup_db_BANG__STAR_$fn__24900.invoke(db.clj:352) ~[?:?]                                                                                                                                                                              
```

**To Reproduce**
Configure PostgreSQL to require client certificate auth, and then launch MB with a `MB_DB_CONNECTION_URI` with the appropriate configuration options.

**Expected behavior**

MB should be able to connect.

**Information about your Metabase Installation:**

Reproduced on the 0.37.x releases. Likely affects all version.

**Comment by tocbfisher** [comment]
Hi; I'm coincidentally trying to migrate my h2 db to a remote postgres db at the moment. Am I correct in understanding that as of right now it is not possible to connect to an sslmode=require server? I'm getting the `FATAL: no pg_hba.conf entry` error.

**Comment by robdaemon** [comment]
@tocbfisher No, `sslmode=require` is hardcoded for SSL connections currently, so that should not be the issue you're facing.

**Comment by tocbfisher** [comment]
@robdaemon Thanks. You're right; my problem was unrelated. I had to add `?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory` to my connection URI. This was very annoying to troubleshoot, but ultimately it seems to be working.

**Comment by camsaul** [comment]
Fixed by #13797
